src/vhost_kern/vhost_binding.rs

```diff
@@ -0,0 +1,405 @@
+// Copyright (C) 2019 Alibaba Cloud Computing. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 or MIT
+//
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+/* Auto-generated by bindgen then manually modified for simplicity */
+
+#![allow(non_upper_case_globals)]
```

Comment:
This is being discussed on the ML, and this is not clear yet, but it seems you will have to put those bindings in a separate crate.

Reply:
I have replied to the ML thread. For the vhost case, I do feel it's better to include the binding into the crate with manual modifications.

Reply:
@sameo @bonzini @andreeaflorescu @zachreizner need feedback here :)

Reply:
From the coverage test perspective is better to have the bindings in a different crate. In Firecracker we are excluding auto-generated files when computing the coverage. If we have them in the same crate it means we will have to make changes to the test_coverage.py so that it is more  flexible and we can exclude independent files depending on the repository being tested. The context is the plan to have common integration tests in one repository and all other repositories to consumes them.

But if we don't use most of the things that are auto-generated then maybe we should just have a minimal vhost_bindings file inside the repsository. I don't have any preference between these two.

Reply:
Good point about the test_coverage.py. Could we solve this issue by adding more test cases?